diag_manager/fms_diag_buffer.F90

```diff
@@ -211,142 +198,62 @@ subroutine set_buffer_id(this, id)
 
 !> Remaps 0-5d data buffer from the given object onto a 5d array and sets it in the type
 subroutine remap_buffer(this)
-  class(fmsDiagBuffer_class), intent(inout) :: this !< any dimension buffer object
-  integer                   :: buff_bounds(5)
+  class(fmsDiagBuffer_class), target, intent(inout) :: this !< any dimension buffer object
```

Comment:
The buffer updater (the "averaging functions") in send_data_3d assumed the buffers are 4D (3D space + sample). Did we decide to move to 5D, or is this remapping for another purpose?